### DISCUSSION TITLE
Warning about  renamed colours

### DISCUSSION THREAD

**Discussion by ricardomadeira93** [tailwindlabs/tailwindcss#13817] [Help]
I have a function that gets all tailwind colors, and then i map them to my component, but on the console i have an warning towards renamed colors.

```jsx
import * as tailwindColors from "tailwindcss/colors"

type ColorOption = {
  label: string
  value: string
  color: string
}



const colorOptions: ColorOption[] = []
const addedColors: Set<string> = new Set()

Object.entries(tailwindColors).forEach(([color, shades]) => {
  if (color === "default" || typeof shades !== "object") return

  Object.entries(shades).forEach(([shade, value]) => {
    const capitalizedLabel = `${color.charAt(0).toUpperCase()}${color.slice(1)} ${shade}`
    if (!addedColors.has(value as string)) {
      colorOptions.push({
        label: capitalizedLabel,
        value,
        color: value as string,
      })
      addedColors.add(value as string)
    }
  })
})

export { colorOptions }
```
```tsx
import { tailwindConfig, typographyConfig } from "@prologue/shared"
import typography from "@tailwindcss/typography"
import { Config } from "tailwindcss"

const config: Config = {
  presets: [tailwindConfig],
  darkMode: "class",
  content: [
    "./index.html",
    "./src/**/*.{js,ts,jsx,tsx}",
    "../ui/src/**/*.{js,ts,jsx,tsx}",
  ],
  theme: {
    extend: {
      typography: typographyConfig,
    },
  },
  plugins: [typography],
}

export default config
```

How can i remove the warnings?

**Comment by wongjn** [comment]
Consider avoiding accessing the properties that emit the warning:

https://github.com/tailwindlabs/tailwindcss/blob/ff6f085da2afe4149ab2791b8b6b74836dbbba9f/src/public/colors.js#L302-L321

**Reply by ricardomadeira93** [reply]
Should that be on tailwind.config.ts?

**Reply by wongjn** [reply]
No, it'd be in the function where you access `tailwindcss/colors`.

**Reply by wongjn** [reply]
> Got new warnings with it.
>
> ```tsx
> import * as tailwindColors from "tailwindcss/colors"
>  
> type ColorOption = {
>   label: string
>   value: string
>   color: string
> }
>  
> const colorOptions: ColorOption[] = []
> const addedColors: Set<string> = new Set()
>  
> const colorNameMapping = {
>   lightBlue: { version: "v2.2", newColor: "sky" },
>   warmGray: { version: "v3.0", newColor: "stone" },
>   trueGray: { version: "v3.0", newColor: "neutral" },
>   coolGray: { version: "v3.0", newColor: "gray" },
>   blueGray: { version: "v3.0", newColor: "slate" },
> }
>  
> const warn = ({ version, from, to }) => {
>   console.warn(
>     `Warning: '${from}' is deprecated since ${version}. Use '${to}' instead.`
>   )
> }
>  
> Object.entries(tailwindColors).forEach(([color, shades]) => {
>   if (color === "default" || typeof shades !== "object") return
>  
>   // Check if the color is in the mapping object
>   if (colorNameMapping[color]) {
>     const { version, newColor } = colorNameMapping[color]
>     warn({ version, from: color, to: newColor })
>     color = newColor
>   }
>  
>   Object.entries(shades).forEach(([shade, value]) => {
>     const capitalizedLabel = `${color.charAt(0).toUpperCase()}${color.slice(1)} ${shade}`
>     if (!addedColors.has(value as string)) {
>       colorOptions.push({
>         label: capitalizedLabel,
>         value,
>         color: value as string,
>       })
>       addedColors.add(value as string)
>     }
>   })
> })
>  
> export { colorOptions }
> Warning: 'blueGray' is deprecated since v3.0. Use 'slate' instead.
> ```

I'm not sure why you did what you did, but yes, that would be expected. As per my initial comment, consider avoiding accessing the deprecated properties.

**Reply by ricardomadeira93** [reply]
How could i avoid accessing them?

**Reply by wongjn** [reply]
You'd probably want to avoid the `Object.entries(tailwindColors)` call, as this would access all the properties, including the deprecated ones.

**Reply by ricardomadeira93** [reply]
Will give it a try, and let you know how it went!
